### PR TITLE
Fix blank screen when signing in with Expensifail account

### PR DESCRIPTION
**Pull request by luacmartins**
### Details
1. A 405 response means that the account needs to be validated and we should let users move to `ResendValidationForm`.
3. We display Account errors in the form, instead of Session errors. This PR changes the error set in `createAccount`

### Fixed Issues
$ https://github.com/Expensify/App/issues/8320

### Tests
1. On the login page, enter a brand new expensifail email
2. Click on Continue
4. Verify that you see the resend link view

- [X] Verify that no errors appear in the JS console

#### Contributor (PR Author) Checklist
- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] iOS / native
    - [x] Android / native
    - [x] iOS / Safari
    - [x] Android / Chrome
    - [x] MacOS / Chrome
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there’s a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [ ] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [ ] I verified that comments were added to code that is not self explanatory
    - [ ] I verified that any new or modified comments were clear, correct English, and explained “why” the code was doing something instead of only explaining “what” the code was doing.
    - [ ] I verified any copy / text shown in the product was added in all `src/languages/*` files
    - [ ] I verified any copy / text that was added to the app is correct English and approved by marketing by tagging the marketing team on the original GH to get the correct copy.
    - [ ] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named “index.js”. All platform-specific files are named for the platform the code supports as outlined in the README.
    - [ ] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/STYLE.md#jsdocs)) were followed
- [ ] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [ ] If a new component is created I verified that:
    - [ ] A similar component doesn't exist in the codebase
    - [ ] All props are defined accurately and each prop has a `/** comment above it */`
    - [ ] Any functional components have the `displayName` property
    - [ ] The file is named correctly
    - [ ] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [ ] The only data being stored in the state is data necessary for rendering and nothing else
    - [ ] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [ ] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [ ] All JSX used for rendering exists in the render method
    - [ ] The component has the minimum amount of code necessary for its purpose and it is
- [ ] If a new CSS style is added I verified that:
    - [ ] A similar style doesn’t already exist
    - [ ] The style can’t be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [ ] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.

<details>
<summary><h4>PR Reviewer Checklist</h4></summary>

- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [ ] I verified the steps for local testing are in the `Tests` section
    - [ ] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [ ] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [ ] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [ ] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I verified tests pass on **all platforms** & I tested again on:
    - [ ] iOS / native
    - [ ] Android / native
    - [ ] iOS / Safari
    - [ ] Android / Chrome
    - [ ] MacOS / Chrome
    - [ ] MacOS / Desktop
- [ ] I verified there are no console errors (if there’s a console error not related to the PR, report it or open an issue for it to be fixed)
- [ ] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [ ] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [ ] I verified that comments were added to code that is not self explanatory
    - [ ] I verified that any new or modified comments were clear, correct English, and explained “why” the code was doing something instead of only explaining “what” the code was doing.
    - [ ] I verified any copy / text shown in the product was added in all `src/languages/*` files
    - [ ] I verified any copy / text that was added to the app is correct English and approved by marketing by tagging the marketing team on the original GH to get the correct copy.
    - [ ] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named “index.js”. All platform-specific files are named for the platform the code supports as outlined in the README.
    - [ ] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/STYLE.md#jsdocs)) were followed
- [ ] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [ ] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/PR_REVIEW_GUIDELINES.md)
- [ ] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [ ] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [ ] If a new component is created I verified that:
    - [ ] A similar component doesn't exist in the codebase
    - [ ] All props are defined accurately and each prop has a `/** comment above it */`
    - [ ] Any functional components have the `displayName` property
    - [ ] The file is named correctly
    - [ ] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [ ] The only data being stored in the state is data necessary for rendering and nothing else
    - [ ] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [ ] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [ ] All JSX used for rendering exists in the render method
    - [ ] The component has the minimum amount of code necessary for its purpose and it is broken down into smaller components in order to separate concerns and functions
- [ ] If a new CSS style is added I verified that:
    - [ ] A similar style doesn’t already exist
    - [ ] The style can’t be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [ ] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [ ] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.

</details>

### QA Steps
Same as testing steps.

- [ ] Verify that no errors appear in the JS console

#### Web

https://user-images.githubusercontent.com/22219519/160199687-e96b2f4d-02b0-4e41-a03e-35ce81537edd.mov

#### Mobile Web

https://user-images.githubusercontent.com/22219519/160199718-6800844b-09db-4a69-a571-76edb2e00b1d.mov

#### Desktop

https://user-images.githubusercontent.com/22219519/160199741-8c141920-d37d-477b-af23-bb13734a191f.mov

#### iOS

https://user-images.githubusercontent.com/22219519/160199758-b5ba5865-f527-481c-96a9-842a827bb05e.mov

#### Android

https://user-images.githubusercontent.com/22219519/160199784-23978233-b1b3-4a6d-8db0-4e1e83976c41.mov

### CODE REVIEW

**Comment by marcaaron** [comment]
>When account creation fails, we set login to null [here](https://github.com/Expensify/App/blob/b698932b4cc33188cbdb15ab7c14dcb4fb37543c/src/libs/actions/Session/index.js#L67) which triggers the following console error and results in a blank screen.

Are we handling the failing account creation? What jsonCode does it send?

**Comment by luacmartins** [comment]
Ready for review!

File: src/libs/actions/Session/index.js [hunk 1]
```diff
@@ -50,21 +50,18 @@ function setSuccessfulSignInData(data) {
...
             return;
         }
 
-        let errorMessage = response.message || `Unknown API Error: ${response.jsonCode}`;
-        if (!response.message && response.jsonCode === 405) {
-            errorMessage = 'Cannot create an account that is under a controlled domain';
-        }
-        Onyx.merge(ONYXKEYS.SESSION, {error: errorMessage});
         Onyx.merge(ONYXKEYS.CREDENTIALS, {login: null});
+        Onyx.merge(ONYXKEYS.ACCOUNT, {error: response.message || `Unknown API Error: ${response.jsonCode}`});
```
Comment (marcaaron) [inline src/libs/actions/Session/index.js:64]: Just to add some context to this change... When we set the login to `null` we get redirected back to the `LoginForm`? And we are handling `account` not `session` errors there?

File: src/pages/signin/ChangeExpensifyLoginLink.js [hunk 2]
```diff
@@ -27,28 +28,31 @@ const defaultProps = {
...
-        <TouchableOpacity
-            style={[styles.link]}
-            onPress={Session.clearSignInData}
-            underlayColor={themeColors.componentBG}
-        >
-            <Text style={[styles.link]}>
-                {props.translate('common.goBack')}
-                {'.'}
+const ChangeExpensifyLoginLink = (props) => {
+    const login = lodashGet(props.credentials, 'login', '');
```
Comment (marcaaron) [inline src/pages/signin/ChangeExpensifyLoginLink.js:32]: question: Is this change still necessary? If the `login` is not in `credentials` and we are here then this is a bug.

**Review by marcaaron — CHANGES_REQUESTED** [review]

File: src/libs/actions/Session/index.js (hunk 1 above)
Comment (luacmartins) [reply src/libs/actions/Session/index.js:64]: Correct!

File: src/pages/signin/ChangeExpensifyLoginLink.js (hunk 2 above)
Comment (luacmartins) [reply src/pages/signin/ChangeExpensifyLoginLink.js:32]: I think it's better to have a fallback for `credentials.login` to prevent crashing the app. In this case, the user would still be redirected to the username step and we would display an error message.

**Comment by aldo-expensify** [comment]
is this the right message for validation when it is a domain-controller account?

**Comment by luacmartins** [comment]
Hmm that's the message that we've been displaying in NewDot this whole time, but I think that the message in OldDot is different. Not sure which one is the correct one TBH, so I kept the current one.

File: src/pages/signin/ChangeExpensifyLoginLink.js (hunk 2 above)
Comment (marcaaron) [reply src/pages/signin/ChangeExpensifyLoginLink.js:32]: **counterpoint:** If the app must crash we have an error boundary for it to hit. It's better to do that than to show content that requires a prop. It might even be _better_ to let the error boundary get hit as it alerts us of a problem by logging (instead of a confusing situation like rendering empty strings instead of required content).

**thought:** maybe the error boundary should also do `ENSURE_BUGBOT` so we can know about and fix stuff right away when it crashes instead of showing broken pages with required variables that we have gave defaults because we were afraid of crashes.

Comment (luacmartins) [reply src/pages/signin/ChangeExpensifyLoginLink.js]: Fair point. I reverted these changes.

**Comment by luacmartins** [comment]
Updated!

**Review by marcaaron — APPROVED** [review]
